nixos/modules/services/misc/paperless.nix

```
@@ -291,6 +295,72 @@ in
       };
     };
 
+    # The following service downloads data for NLTK before the scheduler is
+    # started. Theoretically, this should be possible in an ExecStartPre
+    # directive of service paperless-scheduler. But even with the '+' prefix,
+    # some restrictions were still applied to the download script. Those
+    # restrictions prevented the script from downloading any data. This might be
+    # a bug in systemd. But anyway, moving this task to its own unit has also
+    # security advantages. While commands in a ExecStartPre directive do not
+    # have any restrictions when they are prefixed with '+' (theoretically), we
+    # can put as many restrictions as possible on this service.
+    systemd.services.paperless-download-nltk-data = {
+      wantedBy = [ "paperless-scheduler.service" ];
+      before = [ "paperless-scheduler.service" ];
+      script = let pythonWithNltk = pkg.python.withPackages (ps: [ ps.nltk ]); in
+        ''
+          test -d "${nltkDir}" || mkdir "${nltkDir}"
```

```suggestion
          mkdir -p "${nltkDir}"
```

Should accomplish the same, albeit simpler?

---

Your comment prompted me to have another look at this. I have updated my pull request. The NLTK directory is now created via systemd.tmpfiles.rules, like the other directories.